packages/core-plugin-api/src/apis/definitions/AlertApi.ts

```diff
@@ -26,6 +26,7 @@ export type AlertMessage = {
   message: string;
   // Severity will default to success since that is what material ui defaults the value to.
   severity?: 'success' | 'info' | 'warning' | 'error';
+  transient?: boolean;
```

Comment:
I'm very torn about this tbh. Do you have some examples of where it's clear-cut that you'd use this? If we add this I would somewhat expect there to be quite a few places within plugins in this repo where we should be setting it.

Another direction I feel we could go with this is to implement it all in the `AlertDisplay` without any additions to the API. The `AlertDisplay` is a very low effort early implementation by me tbh, and I was intending for much more powerful displays to be implemented over time. It might for example be that it pops up for a short amount of time somewhere, but then you can access past alerts through the sidebar e.g.

I feel there's some significant overlap with #10652 as well. Ofc the `AlertApi` in general already has that, but if we also start building out more capabilities of it then I feel #10652 should be taken into consideration more carefully.

Reply:
Ok, where the need came from is a plugin we created to allow people to create users and groups (we looked at all the options and this was the best path for us). When a user or group is saved it will trigger the success alert but what we noticed is that they stack up and you have to clear them all by hand. The feedback from testers and endusers was they didn't like it but wanted the alerts.

Could you expand on the idea of implementing it all in the `AlertDisplay`, I'm not totally following? Are you saying we don't add the `transient` property and all alerts close after a timeout?

Yes, #10652 is for sure the best final state. My goal was just a small interim change while we wait on the completion of #10652.

Reply:
Yep, let's add it to the API!